game/assets/scripts/dialogue/dialogue_player.lua

```diff
@@ -1,76 +1,81 @@
-local narrator = require("narrator/narrator")
-local story
-local ui
-local currentChoice = 1
-local maxChoices = 1
+DialoguePlayer = class("DialoguePlayer")
 
-exports = {
-	story = ""
-}
+function DialoguePlayer:initialize(entity)
+    self.exports = {
+		story = ""
+	}
+end
 
-function onBegin(entity)
-	local book = narrator.parseFile(exports.story)
-	story = narrator.initStory(book)
-	story:begin()
-	ui = entity:getUI()
-	getDocument():Show()
+function DialoguePlayer:begin(entity)
+	self.narrator = require("narrator/narrator")
+	local book = self.narrator.parseFile(self.exports.story)
+	self.story = self.narrator.initStory(book)
+	self.story:begin()
+	self.ui = entity:getUI()
+	self.currentChoice = 1
+	self.maxChoices = 1
+	self:getDocument():Show()
+	print("reahed")
 end
 
-function getDocument()
-	return rmlui.contexts["default"].documents[ui:getDocumentID()]
+function DialoguePlayer:update(entity, delta)
+	if self.story:canContinue() then
+		local paragraphs = self.story:continue()
+		self:presentParagraphs(paragraphs)
+		if self.story:canChoose() then
+			self:presentChoices(self.story:getChoices())
+		end
+	end
+
+	if self.story:canChoose() then
+		if RTX.Input.WasPressed("Up") then
```

Comment:
Will also need to configure actions here like the `Up` action

Reply:
done.